src/controllers/comment.js

```diff
@@ -124,16 +124,63 @@ class Comment {
       });
 
       if (!postComment) {
-        return response.status(500).send(errorResponse(['Unable to post comment']));
+        return response.status(500).json(errorResponse(['Unable to post comment']));
       }
 
-      return response.status(201).send({
-        comment: {
-          body: commentBody
-        }
-      });
+      return response.status(201).json({ comment: { body: commentBody } });
+    } catch (error) {
+      return response.status(500).json(errorResponse(['Server error: Failed to post comment']));
+    }
+  }
+
+  /**
+   * @description - Update/Edit a comment by ID
+   *
+   * @static
+   * @param {object} request - Request sent to the router
+   * @param {object} response - Response sent from the controller
+   *
+   * @returns {object} - object representing response message
+   *
+   * @memberof Comment
+   */
+  static async update(request, response) {
+    const { commentId = '' } = request.params;
+
+    if (!commentId) return response.status(400).json(errorResponse(['Please provide ID of comment to update']));
+    if (!parseInt(commentId, 10)) return response.status(400).json(errorResponse(['Please enter a valid comment ID']));
+
+    try {
+      const { payload } = request.userData;
+      const { id: userId } = payload;
+
+      const existingComment = await CommentModel.findByPk(commentId);
+
+      if (!existingComment) return response.status(404).json(errorResponse(['Comment not found']));
+      if (userId !== existingComment.userId) return response.status(401).json(errorResponse(['Not allowed']));
```

Comment:
Great Job @codeshifu. But don't you think this status code should be `403` and not `401`?

Reply:
Thanks, @Johnsonojo for noticing this. It has now been resolved.